Server/core/CThreadCommandQueue.h

```diff
@@ -25,10 +25,8 @@ class CThreadCommandQueue
     void                Process         ( bool& bRequestedQuit, class CModManagerImpl* pModManager );
 
 private:
-    std::list < std::string >       m_Commands;
-    CCriticalSection                m_Critical;
+    std::queue < std::string >  m_Commands;

```

Comment:
Is the use of `std::queue` actually a contributing factor in fixing the crash?  (adding this is a good idea though!)